### PR TITLE
`Exam mode`: Update problem statement content only if there is a new change

### PR DESCRIPTION
**Pull request by kaancayli**
#### General

- [x] This is a small issue that I tested locally and was confirmed by another developer on a test server.
- [x] Language: I followed the [guidelines for inclusive, diversity-sensitive, and appreciative language](https://docs.artemis.cit.tum.de/dev/guidelines/language-guidelines/).
- [x] I chose a title conforming to the [naming conventions for pull requests](https://docs.artemis.cit.tum.de/dev/development-process/#naming-conventions-for-github-pull-requests).

#### Client
- [x] **Important**: I implemented the changes with a very good performance, prevented too many (unnecessary) REST calls and made sure the UI is responsive, even with large data.
- [x] I **strictly** followed the [client coding and design guidelines](https://docs.artemis.cit.tum.de/dev/guidelines/client/).
- [x] I documented the TypeScript code using JSDoc style.
- [x] I added multiple screenshots/screencasts of my UI changes.

### Motivation and Context

After submitting the exam, "Click Show the difference" button can still be seen on the exam overview page and clicking the button does not show any difference. This PR makes sure that after submission, the button is not visible.

### Description

Changed the `BehaviorSubject` to regular `Subject` in `exam-exercise-update.service.ts`

### Exam Mode Testing

Prerequisites:
- 1 Instructor
- 1 Student
- 1 Exam with a Programming Exercise

> [!NOTE]
> Prevent refreshing your browser while following the steps.

1. Log in to Artemis
2. Participate in an exam as a student
3. In another browser / or in a new window using incognito mode, log in to Artemis as an Instructor / Admin.
4. Go to exam management as instructor and edit the problem statement of the programming exercise.
5. As student verify that you can see the "Click Show the difference" button, and clicking the button shows the difference.
6. Make a submission to the programming exercise as a student.
7. After your submission is built and tested, hand in the exam
8. Verify on the exam overview page that "Click Show the difference" button is not visible.

#### Code Review
- [x] Code Review 1
- [x] Code Review 2
#### Exam Mode Test
- [ ] Test 1
- [ ] Test 2

### Screenshots

- Before submission:
Screenshot 2024-04-08 at 23 39 40

- After submission:
Screenshot 2024-04-08 at 23 54 27

### CODE REVIEW

File: src/main/webapp/app/exam/participate/exercises/exam-exercise-update-highlighter/exam-exercise-update-highlighter.component.ts [hunk 1]
```diff
@@ -24,10 +25,20 @@ export class ExamExerciseUpdateHighlighterComponent implements OnInit, OnDestroy
...
+            } else {
+                // No update so hide the component
+                this._isHidden = true;
+            }
         });
     }
 
+    get isHidden(): boolean {
+        return this.previousProblemStatementUpdate === undefined || this._isHidden;
+    }
```
Comment (Strohgelaender) [inline src/main/webapp/app/exam/participate/exercises/exam-exercise-update-highlighter/exam-exercise-update-highlighter.component.ts:40]: We should avoid method calls from the template, so please move this piece directly into the `[hidden]` property.

**Review by Strohgelaender — CHANGES_REQUESTED** [review]

File: src/main/webapp/app/exam/participate/exercises/exam-exercise-update-highlighter/exam-exercise-update-highlighter.component.ts (hunk 1 above)
Comment (kaancayli) [reply src/main/webapp/app/exam/participate/exercises/exam-exercise-update-highlighter/exam-exercise-update-highlighter.component.ts]: Made the necessary changes. Thanks for the review.

**Review by Strohgelaender — DISMISSED** [review]
code

**Review by JohannesWt — DISMISSED** [review]
Code looks good

**Review by julian-christl — APPROVED** [review]
Don't see any such button on TS2, code lgtm

**Review by coolchock — APPROVED** [review]
tested during testing session, works as described 👍

**Review by undernagruzez — APPROVED** [review]
tested during the testing session today, works as expected

**Review by pzdr7 — APPROVED** [review]
Tested in testing session; works as expected

**Review by BaumiCoder — APPROVED** [review]
Successfully tested in the Testing Session :+1:

**Pull request merged**